library/crypto/sha512.c

```diff
@@ -321,6 +324,72 @@ sha384_stream (FILE *stream, void *resblock)
   return 0;
 }
 
+/* Compute SHA512 message digest for bytes read from file descriptor.
+   The resulting message digest number will be written into the 64
+   bytes beginning at RESBLOCK.  */
+int
+sha512_file_ctx (int fd, off_t offset, size_t count, struct sha512_ctx *ctx)
```

Comment:
This task (moving crypto lib to eblob) indirectly connected with mmap elimination.
I will do this in a separate PR (as a separate task), as I mentioned above.